### PR TITLE
[ML] Support for Singularity Job Schema properties like Tier, Priority

### PR DESCRIPTION
**Pull request by TonyJ1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/queue_settings.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from marshmallow import fields, post_load
+from azure.ai.ml.constants._job.job import JobPriorityValues, JobTierNames
+from azure.ai.ml._schema.core.fields import StringTransformedEnum
+from ..core.schema import PathAwareSchema
+
+from .resource_configuration import ResourceConfigurationSchema
+
+
+class QueueSettingsSchema(PathAwareSchema):
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/queue_settings.py:13]: I don't see this schema being added to command/sweep job schema. Is that expected?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py [hunk 2]
```diff
@@ -161,14 +164,14 @@ def __init__(
         environment: Optional[Union[Environment, str]] = None,
         environment_variables: Optional[Dict] = None,
         resources: Optional[JobResourceConfiguration] = None,
+        queue_settings: Optional[QueueSettings] = None,
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:167]: Let also add flattened properties of `job_tier` and `priority` as we discussed.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py [hunk 3]
```diff
@@ -78,6 +86,8 @@ def __ne__(self, other: object) -> bool:
 
     def _merge_with(self, other: "ResourceConfiguration") -> None:
         if other:
+            if other.locations:
+                self.locations = other.locations
             if other.instance_count:
                 self.instance_count = other.instance_count
             if other.instance_type:
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py:93]: Please add some test cases.

**Review by luigiw — CHANGES_REQUESTED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/queue_settings.py (hunk 1 above)
Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/queue_settings.py:11]: The changes will be part of Vijetha's PR or I will send a follow up PR.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py (hunk 2 above)
Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:167]: The changes will be part of Vijetha's PR or I will send a follow up PR.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py (hunk 3 above)
Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py]: Removing the changes to locations from this PR.

**Pull request closed without merging**